torchci/docs/architecture.md

```diff
@@ -0,0 +1,82 @@
+[`issues`]: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#issues
+[GitHub Apps]: https://docs.github.com/en/developers/apps/getting-started-with-apps/about-apps
+[PyTorch Bot]: https://github.com/apps/pytorch-bot
+[`/api/github/webhooks`]: https://github.com/pytorch/test-infra/blob/main/torchci/pages/api/github/webhooks.ts
+[dynamo]: https://rockset.com/docs/amazon-dynamodb/
+
+## Secondary write paths
+There are some parts to the system that don't use the write path described
+above, because GitHub doesn't generate webhooks for them.
+
+Notably, this means that **installing the PyTorch Bot app to repos will not
+enable these write paths**. They must be manually wired up.
+
+#### Build artifacts
+Our build jobs create various artifacts that we want to save to persistent
+storage and use later. For example, most of our build jobs produce binary wheel
+distributions of PyTorch that are later downloaded by our test jobs.
+
+These are uploaded to S3 directly by the GitHub workflows in `pytorch/pytorch`.
+
+#### Logs and log classifications
+Logs are downloaded by the [`github-status-test`] AWS lambda, which receives
+webhooks through a manual connection with the `pytorch/pytorch` repo.
+`github-status-test` then pings another lambda, [`log-classifier`], which
+performs log classification (more detail in the [README]).
+
+[`github-status-test`]: https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/github-status-test?tab=code
+[`log-classifier`]: https://github.com/pytorch/test-infra/tree/main/aws/lambda/log-classifier
+[README]: https://github.com/pytorch/test-infra/blob/main/aws/lambda/log-classifier/README.md
+
+The logic in `github-status-test` could be moved to the `hud.pytorch.org`
+webhooks endpoint, and modified so that all repos that `torchci` is installed
+on benefits from log downloads and classifications, but this is not currently
+implemented. Add @suo as a reviewer on a PR if you want to add it!
+
+#### Test statistics
+See the [README](https://github.com/pytorch/pytorch/tree/master/tools/stats).
+
+### CircleCI results
+CircleCI also has webhooks like GitHub, but they must be configured manually.
+The webhooks are delivered to the [`ossci-circleci-to-s3`] lambda, which is
+synced to Rockset. See the [example config] for `pytorch/vision` webhooks for
+an example.
+
+[`ossci-circleci-to-s3`]: https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/ossci-circleci-to-s3?tab=code
+[example config]: https://app.circleci.com/settings/project/github/pytorch/vision/webhooks/a5ae92ec-d523-4495-be2a-58ab80a255b9?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fpytorch%2Fvision
+
+## Adding a new repo to `torchci`
+To get the basic HUD working, you will need to:
+1. Install the PyTorch Bot app to the repo. This requires admin access to the
```

Comment:
May be we can refer to the poc for Build+CI in https://pytorch.org/docs/master/community/persons_of_interest.html for requests that need admin access, any thought?

Reply:
sure